### PR TITLE
ConstantScore query & boost support

### PR DESCRIPTION
**Pull request by betenkowski**
Boost support added to:
* `TermQuery`,
* `WildcardQuery`,
* `PrefixQuery`.

`ConstantScore` query support.

### CODE REVIEW

File: elasticsearch-core/src/test/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClient6Test.scala [hunk 1]
```diff
@@ -104,8 +104,7 @@ class RestlasticSearchClient6Test extends WordSpec with Matchers with BeforeAndA
       implicit val formats = org.json4s.DefaultFormats
       val docsCount = 10011
       val documents = (1 to docsCount).map(i => Document(s"doc$i", Map("text7" -> "here7")))
-      val bulkInsertResult = restClient.bulkIndex(index, tpe, documents)
-      Await.result(bulkInsertResult, 20.seconds)
+      restClient.bulkIndex(index, tpe, documents).futureValue
```
Comment (michaljmatusiak92) [inline elasticsearch-core/src/test/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClient6Test.scala:107]: Is `20` seconds not enough?

File: elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/dsl/QueryDsl.scala [hunk 2]
```diff
@@ -220,12 +228,11 @@ trait QueryDsl extends DslCommons with SortDsl {
     override def toJson(version: EsVersion): Map[String, Any] = Map(_lte -> value)
   }
 
-  case class WildcardQuery(key: String, value: String, boost: Option[Float] = None) extends Query {
+  case class WildcardQuery(key: String, value: String, boost: Option[Float] = None) extends Query with ValueBoost {
```
Comment (michaljmatusiak92) [inline elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/dsl/QueryDsl.scala:228]: Just to be sure: no `override val value/val value` is needed?

**Review by michaljmatusiak92 — APPROVED** [review]

File: elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/dsl/QueryDsl.scala (hunk 2 above)
Comment (betenkowski) [reply elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/dsl/QueryDsl.scala:228]: No, Scala is smart enough to figure this out.

File: elasticsearch-core/src/test/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClient6Test.scala (hunk 1 above)
Comment (betenkowski) [reply elasticsearch-core/src/test/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClient6Test.scala:107]: It was more about consistency with the rest of the tests than about 20 seconds (and yes, I wasn't determined enough to fix this everywhere).

**Pull request merged**